### PR TITLE
vz-9316 Validating Custom Resource in VZ CLI

### PR DESCRIPTION
**Pull request by AbeHern**
Added logic to validate a Custom Resource when trying to install using the VZ CLI

### CODE REVIEW

**Comment by prasad-shirodkar** [comment]
Please implement an e2e test to verify the behavior.

File: tools/vz/test/testdata/invalidCR.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
```
Comment (bobdonat) [inline tools/vz/test/testdata/invalidCR.yaml:1]: i think this file can be removed

File: tools/vz/test/testdata/validCR.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
```
Comment (bobdonat) [inline tools/vz/test/testdata/validCR.yaml:1]: same comment - i think this file can be removed

File: tools/vz/cmd/install/install.go [hunk 3]
```diff
@@ -221,13 +240,19 @@ func runCmdInstall(cmd *cobra.Command, args []string, vzHelper helpers.VZHelper)
...
 	// Wait for the platform operator to be ready before we create the Verrazzano resource.
 	_, err := cmdhelpers.WaitForPlatformOperator(client, vzHelper, v1beta1.CondInstallComplete, vpoTimeout)
 	if err != nil {
 		return err
 	}
 
+	// Validate Custom Resource if present
+	var errorArray = ValidateCRFunc(cmd, obj, vzHelper)
+	if len(errorArray) != 0 {
+		return fmt.Errorf("was unable to validate the given CR, the following error(s) occurred: \"%v\"", errorArray)
```
Comment (bobdonat) [inline tools/vz/cmd/install/install.go:253]: did this show multiple errors in a readable format

**Comment by bobdonat** [comment]
I do not see the update to go.mod/go.sum for the new dependency on k8s.io/kubectl/pkg/util/openapi

**Review by bobdonat — APPROVED** [review]

**Pull request merged**